x/ecocredit/server/core/msg_toggle_class_allowlist_test.go

```diff
@@ -0,0 +1,59 @@
+package core
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/regen-network/gocuke"
+	"github.com/stretchr/testify/require"
+
+	"github.com/regen-network/regen-ledger/x/ecocredit/core"
+)
+
+type toggleClassCreatorAllowlist struct {
+	*baseSuite
+	err error
+}
+
+func TestToggleClassCreatorAllowlist(t *testing.T) {
+	gocuke.NewRunner(t, &toggleClassCreatorAllowlist{}).Path("./features/msg_toggle_class_allowlist.feature").Run()
+}
+
+func (s *toggleClassCreatorAllowlist) Before(t gocuke.TestingT) {
+	s.baseSuite = setupBase(t)
+}
+
+func (s *toggleClassCreatorAllowlist) AliceAttemptsToToggleClassAllowlistWithProperties(a gocuke.DocString) {
+	var msg *core.MsgToggleCreditClassAllowlist
+
+	err := json.Unmarshal([]byte(a.Content), &msg)
+	require.NoError(s.t, err)
+
+	_, s.err = s.k.ToggleCreditClassAllowlist(s.ctx, msg)
+}
+
+func (s *toggleClassCreatorAllowlist) ExpectClassAllowlistFlagToBeTrue() {
+	res, err := s.stateStore.AllowListEnabledTable().Get(s.ctx)
+	require.NoError(s.t, err)
+
+	require.Equal(s.t, true, res.Enabled)
+}
+
+func (s *toggleClassCreatorAllowlist) ExpectClassAllowlistFlagToBeFalse() {
+	res, err := s.stateStore.AllowListEnabledTable().Get(s.ctx)
+	require.NoError(s.t, err)
+
+	require.Equal(s.t, false, res.Enabled)
+}
```

Comment:
These could be one step function, i.e. `ExpectClassAllowlistFlagSetting(a bool)`